.github/workflows/_cicd.yml

```diff
@@ -0,0 +1,57 @@
+name: _cicd (build and release entry)
+
+on:
+  push:
+    branches:
+      - main
+  release:
+    types:
+      - released
+  workflow_dispatch:
+    inputs:
+      commit_hash:
+        description: 'The SHA of the Git commit to use'
+        required: false
+        default: refs/heads/main
+      core_repo_reference:
+        description: 'The SHA to set the core repo to'
+        required: false
+        default: refs/heads/main
+      config_repo_reference:
+        description: 'The SHA to set the config repo to'
+        required: false
+        default: refs/heads/main
+
+permissions:
+  contents: read
+
+jobs:
+  release-dev:
+    if: github.event_name == 'workflow_dispatch'
+    uses: i-dot-ai/redbox-copilot/.github/workflows/build-and-release.yml@a37b08ab5253a8ba3b592b61cb367c47d022f298
+    with:
+      environment: dev
+      core_infra_repo_reference: ${{ inputs.core_repo_reference }}
+      config_repo_reference: ${{ inputs.config_repo_reference }}
+      commit_hash: ${{ inputs.commit_hash }}
+    secrets: inherit
+
+  release-preprod:
+    if: github.event_name == 'push' && github.ref_name == 'main'
+    uses: i-dot-ai/redbox-copilot/.github/workflows/build-and-release.yml@a37b08ab5253a8ba3b592b61cb367c47d022f298
+    with:
+      environment: preprod
+      core_infra_repo_reference: 'refs/heads/main'
```

Comment:
Nice - I prefer this hard-coding over the `env:` block as I find it expresses intent a lot better 👍

Reply:
I found that you actually can't use `env` defined vars in a `with` block, caused me headaches for ages lol